waiter/src/waiter/settings.clj

```diff
@@ -92,6 +92,7 @@
    ;; TODO port belongs in server-options?
    (s/required-key :port) (s/either schema/positive-int
                                     [schema/positive-int])
+   (s/required-key :profile-config) {schema/non-empty-string schema/profile-definition}
```

Comment:
Why are we making this required? It seems like we should always make new keys optional if possible to avoid breaking our existing config files. I see no harm in the profile config coming back `nil` if it's not set (assuming we used nil-safe accessors—which I think we did here since it only seems to be used as a sequence in this patch).

Reply:
`settings` will already provide a default value (empty map), just like we do for all config we add.

The required is necessary to simplify restructuring in `pc/fnk`

Reply:
Ah, I thought this schema was applied directly to the config file. This makes sense now.